internal/cephfs/cephfs_util.go

```diff
@@ -34,23 +34,28 @@ type CephFilesystemDetails struct {
 	MDSMap MDSMap `json:"mdsmap"`
 }
 
-func getFscID(ctx context.Context, monitors string, cr *util.Credentials, fsName string) (int64, error) {
-	// ceph fs get myfs --format=json
-	// {"mdsmap":{...},"id":2}
-	var fsDetails CephFilesystemDetails
-	err := execCommandJSON(ctx, &fsDetails,
-		"ceph",
-		"-m", monitors,
-		"--id", cr.ID,
-		"--keyfile="+cr.KeyFile,
-		"-c", util.CephConfigPath,
-		"fs", "get", fsName, "--format=json",
-	)
+func (vo *volumeOptions) getFscID(ctx context.Context) (int64, error) {
+	fsa, err := vo.conn.GetFSAdmin()
 	if err != nil {
+		util.ErrorLog(ctx, "could not get FSAdmin, can not fetch filesystem ID for %s:", vo.FsName, err)
 		return 0, err
 	}
 
-	return fsDetails.ID, nil
+	volumes, err := fsa.EnumerateVolumes()
+	if err != nil {
+		util.ErrorLog(ctx, "could not list volumes, can not fetch filesystem ID for %s:", vo.FsName, err)
+		return 0, err
+	}
+
+	for _, vol := range volumes {
+		if vol.Name == vo.FsName {
+			return vol.ID, nil
+		}
+	}
+
+	err = fmt.Errorf("volume %s not found", vo.FsName)
```

Comment:
ErrVolumeNotFound can be passed from here?

Reply:
sure, I can do that